controllers/volsync/vshandler.go

```diff
@@ -767,6 +767,28 @@ func (v *VSHandler) DeleteRD(pvcName string) error {
 	return nil
 }
 
+func (v *VSHandler) DeleteSnapshots(pvcNamespace string) error {
+	// Remove a Snapshot by name that is owned (by parent vrg owner)
+	snapList := &snapv1.VolumeSnapshotList{}
+
+	err := v.listByOwner(snapList, pvcNamespace)
+	if err != nil {
+		return err
+	}
+
+	for i := range snapList.Items {
+		snapshot := snapList.Items[i]
+		// Delete the VolumeSnapshot, log errors with cleanup but continue on
+		if err := v.client.Delete(v.ctx, &snapshot); err != nil {
+			v.log.Error(err, "Error cleaning up VolumeSnapshot", "name", snapshot.GetName())
```

Comment:
Error deleting?

Also, if the user need to do manual cleanup or inspect this error, they also need the namespace - is it included in the log? If not, maybe add it to the log as additional arguments ("namespace", pvcNamespace)?

Reply:
In this specific file we have at least 5 places, where we are failing to delete resource, logging the error and proceeding anyway. All these logs are with name only, without namespace. If we want to change the log, I think it's better to do it for all deletions in order to maintain the same logging behaviour.

Reply:
You are in `processForDeletion`. We should not ignore errors other than `NotFound`. You can continue iterating through the list, but return an error.

Reply:
@BenamarMk Done